pkg/controllers/componentroutesecretsync/component_route_secret_sync_controller.go

```diff
@@ -55,18 +62,41 @@ type componentRouteSecretSyncController struct {
 	ingressLister       configlistersv1.IngressLister
 	secretLister        corev1listers.SecretLister
 	resourceSyncer      resourcesynccontroller.ResourceSyncer
+	operatorClient      v1helpers.OperatorClient
 }
 
+// degradedConditionTypes contains all conditions that this controller manages
+var degradedConditionTypes = sets.NewString(
+	"IngressComponentRouteControllerDegraded",
+)
+
 func (c *componentRouteSecretSyncController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
 	ingress, err := c.ingressLister.Get("cluster")
 	if err != nil {
-		return err
+		return common.UpdateControllerConditions(c.operatorClient, degradedConditionTypes, []operatorv1.OperatorCondition{
+			{
+				Type:    "IngressComponentRouteControllerDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "UnableToRetriveClusterIngressConfig",
+				Message: fmt.Sprintf("Unable to get ingress.config.openshift.io/cluster resource: %s", err.Error()),
+			}})
```

Comment:
this gets handled by the controller from the controller factory, I don't think this is necessary at this point.